### PR TITLE
Remove LocationXY usage on windows and actions

### PR DESCRIPTION
**Pull request by tupaschoal**
Helps #10267

### CODE REVIEW

File: src/openrct2-ui/windows/Map.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ static rct_widget window_map_widgets[] = {
 
 // used in transforming viewport view coordinates to minimap coordinates
 // rct2: 0x00981BBC
-static constexpr const LocationXY16 MiniMapOffsets[] = {
+static constexpr const CoordsXY MiniMapOffsets[] = {
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Map.cpp]: The idea of the Coords and TileCoords structures is that you can be sure about what's stored inside. 

These are not big coords or small coords, and thus shouldn't use CoordsXY or TileCoordsXY.

File: src/openrct2-ui/windows/Multiplayer.cpp [hunk 2]
```diff
@@ -300,7 +300,7 @@ static void window_multiplayer_draw_tab_images(rct_window* w, rct_drawpixelinfo*
 static void window_multiplayer_set_page(rct_window* w, int32_t page);
 
 static bool _windowInformationSizeDirty;
-static LocationXY16 _windowInformationSize;
+static CoordsXY _windowInformationSize;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Multiplayer.cpp]: Same as above - not actually Coords, so don't abuse the struct.

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 3]
```diff
@@ -3622,7 +3622,7 @@ void ride_construction_toolupdate_construct(ScreenCoordsXY screenCoords)
         && ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_IS_SHOP))
     {
         TileElement* pathsByDir[4];
-        constexpr sLocationXY8 DirOffsets[4] = { { -1, 0 }, { 0, 1 }, { 1, 0 }, { 0, -1 } };
+        constexpr CoordsXY DirOffsets[4] = { { -1, 0 }, { 0, 1 }, { 1, 0 }, { 0, -1 } };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/RideConstruction.cpp]: These are TileCoordsXY offsets, not CoordsXY.

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 4]
```diff
@@ -1210,14 +1210,14 @@ static void scenery_eyedropper_tool_down(int16_t x, int16_t y, rct_widgetindex w
...
-    uint32_t* parameter_2, uint32_t* parameter_3)
+    int16_t x, int16_t y, uint16_t selected_scenery, CoordsXY& gridPos, uint32_t* parameter_1, uint32_t* parameter_2,
+    uint32_t* parameter_3)
 {
     rct_window* w = window_find_by_class(WC_SCENERY);
 
     if (w == nullptr)
     {
-        *grid_x = LOCATION_NULL;
+        gridPos.x = LOCATION_NULL;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TopToolbar.cpp]: Use setNull().

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 5]
```diff
@@ -1489,11 +1486,11 @@ static void sub_6E1F34(
...
             }
 
-            if (*grid_x == LOCATION_NULL)
+            if (gridPos.x == LOCATION_NULL)
                 return;
 
-            *grid_x &= 0xFFE0;
-            *grid_y &= 0xFFE0;
+            gridPos.x &= 0xFFE0;
+            gridPos.y &= 0xFFE0;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TopToolbar.cpp]: ToTileStart()

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 6]
```diff
@@ -127,9 +127,9 @@ static void window_track_place_draw_mini_preview_track(
     TrackDesign* td6, int32_t pass, CoordsXY origin, CoordsXY& min, CoordsXY& max);
 static void window_track_place_draw_mini_preview_maze(
     TrackDesign* td6, int32_t pass, CoordsXY origin, CoordsXY& min, CoordsXY& max);
-static LocationXY16 draw_mini_preview_get_pixel_position(int16_t x, int16_t y);
-static bool draw_mini_preview_is_pixel_in_bounds(LocationXY16 pixel);
-static uint8_t* draw_mini_preview_get_pixel_ptr(LocationXY16 pixel);
+static CoordsXY draw_mini_preview_get_pixel_position(CoordsXY pos);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp]: See first remark.

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 7]
```diff
@@ -707,17 +705,17 @@ static void window_track_place_draw_mini_preview_maze(
     }
 }
 
-static LocationXY16 draw_mini_preview_get_pixel_position(int16_t x, int16_t y)
+static CoordsXY draw_mini_preview_get_pixel_position(CoordsXY pos)
 {
-    return { (int16_t)(80 + ((y / 32) - (x / 32)) * 4), (int16_t)(38 + ((y / 32) + (x / 32)) * 2) };
+    return { (80 + ((pos.y / 32) - (pos.x / 32)) * 4), (38 + ((pos.y / 32) + (pos.x / 32)) * 2) };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp]: Convert to TileCoordsXY first.

File: src/openrct2-ui/windows/Map.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/Map.cpp]: I've reverted these changes.

File: src/openrct2-ui/windows/Multiplayer.cpp (hunk 2 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/Multiplayer.cpp]: I dropped this commit.

File: src/openrct2-ui/windows/TopToolbar.cpp (hunk 4 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/TopToolbar.cpp]: This is `LOCATION_NULL` which is a `(int16_t)(uint16_t)0x8000` while `setNull` uses `COORDS_NULL` -> `int32_t = 0xFFFF8000`
Should I still use it?

File: src/openrct2-ui/windows/TrackDesignPlace.cpp (hunk 6 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/TrackDesignPlace.cpp]: I've dropped the whole commit on `TrackDesignPlace´

File: src/openrct2-ui/windows/TopToolbar.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/TopToolbar.cpp]: Yes, but make sure that anything that tries setting the gridPos to null properly uses setNull().

File: src/openrct2-ui/windows/Map.cpp [hunk 8]
```diff
@@ -1635,8 +1635,7 @@ static void map_window_set_pixels(rct_window* w)
     int32_t x = 0, y = 0, dx = 0, dy = 0;
 
     int32_t pos = (_currentLine * (MAP_WINDOW_MAP_SIZE - 1)) + MAXIMUM_MAP_SIZE_TECHNICAL - 1;
-    LocationXY16 destinationPosition = { static_cast<int16_t>(pos % MAP_WINDOW_MAP_SIZE),
-                                         static_cast<int16_t>(pos / MAP_WINDOW_MAP_SIZE) };
+    CoordsXY destinationPosition = { pos % MAP_WINDOW_MAP_SIZE, pos / MAP_WINDOW_MAP_SIZE };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Map.cpp]: Also revert this change.

File: src/openrct2-ui/windows/ViewClipping.cpp [hunk 9]
```diff
@@ -303,8 +303,8 @@ static void window_view_clipping_update(rct_window* w)
     if (_toolActive && !window_view_clipping_tool_is_active())
     {
         _toolActive = false;
-        gClipSelectionA = _previousClipSelectionA;
-        gClipSelectionB = _previousClipSelectionB;
+        gClipSelectionA = { static_cast<uint8_t>(_previousClipSelectionA.x), static_cast<uint8_t>(_previousClipSelectionA.y) };
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/ViewClipping.cpp]: Why are all these casts necessary?

Comment (tupaschoal) [reply src/openrct2-ui/windows/ViewClipping.cpp]: `gClipSelectionA` is a `LocationXY8` struct, so I'm making the narrowing explicit.

File: src/openrct2-ui/windows/Map.cpp (hunk 8 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/Map.cpp]: Dropped the entire `Map` commit then.

File: src/openrct2-ui/windows/TopToolbar.cpp (hunk 4 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/TopToolbar.cpp]: Done.

File: src/openrct2-ui/windows/ViewClipping.cpp (hunk 9 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/ViewClipping.cpp]: Looks like gClipSelectionA / B contains tile coordinates - can you modify it to contain TileCoords?

File: src/openrct2/world/Location.hpp [hunk 10]
```diff
@@ -160,6 +160,11 @@ struct CoordsXY
     {
         return x == COORDS_NULL;
     };
+
+    void setNull()
```
Comment (ZehMatt) [inline src/openrct2/world/Location.hpp]: Our member functions are usually starting with a capital letter, any reason those null getter/setter are different?

Comment (tupaschoal) [reply src/openrct2/world/Location.hpp]: I noticed that too, the pair of `isNull/setNull` on `TileCoordsXY` use lowercase, so I just kept consistency here. I can do another PR later converting them all

Comment (Gymnasiast) [reply src/openrct2/world/Location.hpp]: An oversight. I do think that renaming them in this PR would get a bit messy, so I'd prefer if it was done in another.

File: src/openrct2-ui/windows/ViewClipping.cpp (hunk 9 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/ViewClipping.cpp]: I modified `_previousClipSelectionA` and `_previousClipSelectionB` to be `TileCoordsXY`. I didn't touch `gClipSelection(A|B)` because those are bigger changes that even involve some comparisons to sprite coords and I had let them out of this PR on purpose.
